lib/ipmi_backend_utils.pm

```diff
@@ -528,4 +528,22 @@ sub set_grub_terminal_and_timeout {
     }
 }
 
+sub reconnect_when_ssh_console_broken {
+
+    # Switch to sol console to check serial console output
+    # It is useful in the case of host crash and reboot
+    record_info("WARN", "ssh connection is broken and switch to SOL console", result => 'fail');
+    select_console 'sol', await_console => 0;
+    save_screenshot;
+    wait_still_screen 5;
```

Comment:
I am afraid test will spin around this ```wait_still_screen 5``` for a quite while if there is continuous output on sol console. Do you really need "wait_still_screen 5``` ?

Reply:
If there is continuous output, it means the host is still in booting process. the wait will be neccessary and it does't waste time.

Reply:
I do not think you need

wait_still_screen 5;
check_screen("linux-login", 120);

You can use

check_port_state

in lib/virt_autotest/utils.pm
The advantage of using this is there will no waste of time on wait for something uncertainty. check_port_state gives you the certainty. More clear logic and code. PRJ2 host offline upgrade already adopted this method which looks much better.